### ISSUE TITLE
Affine Registration runtime error

### ISSUE THREAD

**Issue by andrewsilberfeld** [BIOP/ijp-imagetoatlas#113]
Hello,

This is a fabulous program and I love what has been done so far! I find it really user friendly and powerful to use. I'm wondering if you can help with a runtime error I get simply when running an affine registration on a single slice (or many slices). The error says: "Elastix2DAFfineRegistration:java.lang.RuntimeException: Module threw error".  Any idea what causes this? I tried running on server but it seemed like had finished after 20 minutes though I'm not sure if that is normal or not. this is my first time attempting anything with registration beside manual.

This looks like the cosole output from this command:
NR Converting:ch.epfl.biop.wrappers.elastix.RegParamAffine_Fast->ch.epfl.biop.wrappers.elastix.RegistrationParameters
R Converting:ch.epfl.biop.wrappers.elastix.RegistrationParameters->java.io.File
Field UseBinaryFormatForTransformationParameters not found in itc.transforms.elastix.ElastixAffineTransform class.
java.util.concurrent.ExecutionException: java.lang.RuntimeException: Module threw error
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:192)
	at ch.epfl.biop.registration.sourceandconverter.affine.Elastix2DAffineRegistration.register(Elastix2DAffineRegistration.java:129)
	at ch.epfl.biop.atlas.aligner.SliceSources.performRegistration(SliceSources.java:439)
	at ch.epfl.biop.atlas.aligner.SliceSources.runRegistration(SliceSources.java:467)
	at ch.epfl.biop.atlas.aligner.RegisterSlice.run(RegisterSlice.java:66)
	at ch.epfl.biop.atlas.aligner.SliceSources.lambda$enqueueRunAction$1(SliceSources.java:506)
	at java.util.concurrent.CompletableFuture.uniApply(CompletableFuture.java:602)
	at java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:577)
	at java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:443)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
Caused by: java.lang.RuntimeException: Module threw error
	at org.scijava.module.ModuleRunner.call(ModuleRunner.java:130)
	at org.scijava.module.ModuleRunner.call(ModuleRunner.java:63)
	at org.scijava.thread.DefaultThreadService.lambda$wrap$2(DefaultThreadService.java:225)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.NoClassDefFoundError: itc/converters/ElastixAffine2DToAffineTransform3D
	at ch.epfl.biop.sourceandconverter.register.Elastix2DAffineRegister.run(Elastix2DAffineRegister.java:189)
	at ch.epfl.biop.bdv.command.register.Elastix2DAffineRegisterCommand.run(Elastix2DAffineRegisterCommand.java:85)
	at org.scijava.command.CommandModule.run(CommandModule.java:196)
	at org.scijava.module.ModuleRunner.run(ModuleRunner.java:163)
	at org.scijava.module.ModuleRunner.call(ModuleRunner.java:124)
	... 6 more
Caused by: java.lang.ClassNotFoundException: itc.converters.ElastixAffine2DToAffineTransform3D
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 11 more
[ERROR] Multipositioner : Elastix2DAffineRegistration: java.lang.RuntimeException: Module threw error

**Comment by andrewsilberfeld** [comment]
spline registration runs successfully but not affine

**Comment by NicoKiaru** [comment]
Hi @andrewsilberfeld and thanks for your message!

The server registration is quite buggy and unless I find some unexpected time resources, I will probably get rid of it. Can you try to install elastix locally ? https://biop.github.io/ijp-imagetoatlas/installation.html

However I'm not sure the issue is this...

Can you go in Fiji to `Help>Update...> Advanced mode` and look at this image_transform_converters file:

![image](https://user-images.githubusercontent.com/20223054/143004582-169a49a2-c9e9-4375-8044-475ef821dd8c.png)

What the version number ? Do you even have such a file (make sure to check `view all files`)?

**Comment by andrewsilberfeld** [comment]
The affine registration error was using the local elastix installation and the server had just never finished when I ran it over night.

I deleted fiji and redownloaded it from scratch and surprisingly it fixed the problem. The image-transform-converts.jar was version 0-1-1.jar from July 2020 before deleting fiji. Upon redownload, it's now 0-1-5 like your picture. 

Thanks for the prompt response and I'm super excited to have this tool!
